rrrspec-client/lib/rrrspec/client/slave_runner.rb

```diff
@@ -57,9 +56,12 @@ def work
           @timeout = TASKQUEUE_TASK_TIMEOUT
           trial = Trial.create(task, @slave)
 
-          @rspec_runner.reset
+          task_file_ext = File.extname(task.spec_file)
+          runner = RRRSpec::Registry.create_runner(task_file_ext)
```

Comment:
This code assumes the underlying factory to return the same instance always. Factories in general return a fresh new instance, and this assumption does not align with that concept.